File: Robust/src/ClassLibrary/Math.java
public class Math {

  public static double setPI() {
    double PI = 3.14159265358979323846;
    return PI;
  }

  public static double fabs(double x) {
    if (x < 0) {
      return -x;
    } else {
      return x;
    }
  }

  public static float abs(float a) {
    if (a<0)
      return -a;
    else return a;
  }

  public static double max(double a, double b) {
    if(a == b)
      return a;
    if(a > b) {
      return a;
    } else {
      return b;
    }
  }

  public static int imax(int a, int b) {
    if(a == b)
      return a;
    if(a > b) {
      return a;
    } else {
      return b;
    }
  }

  public static int imin(int a, int b) {
    if(a == b)
      return a;
    if(a > b) {
      return b;
    } else {
      return a;
    }
  }

  /** sqrt(a^2 + b^2) without under/overflow. **/
  public static double hypot(double a, double b) {
    double r;
    if (fabs(a) > fabs(b)) {
      r = b/a;
      r = fabs(a)*sqrt(1+r*r);
    } else if (b != 0) {
      r = a/b;
      r = fabs(b)*sqrt(1+r*r);
    } else {
      r = 0.0;
    }
    return r;
  }

  public static native double sin(double a);
  public static native double cos(double a);
  public static native double asin(double a);
  public static native double acos(double a);
  public static native double tan(double a);
  public static native double atan(double a);
  public static native double exp(double a);
  public static native double sqrt(double a);
  public static native double log(double a);
  public static native double pow(double a, double b);

  public static native float sinf(float a);
  public static native float cosf(float a);
  public static native float expf(float a);
  public static native float sqrtf(float a);
  public static native float logf(float a);
  public static native float powf(float a, float b);
}

File: Robust/src/Runtime/math.c
#include "runtime.h"
#ifdef MULTICORE
#ifdef RAW
#include "math.h"
#endif
#else
#include "math.h"
#endif
#include "structdefs.h"
#if 0
double CALL11(___Math______cos____D, double ___a___, double ___a___) {
#ifdef MULTICORE
  return 1;
#ifdef RAW
  return cos(___a___);
#endif
#else
  return cos(___a___);
#endif
}

double CALL11(___Math______sin____D, double ___a___, double ___a___) {
#ifdef MULTICORE
  return 1;
#ifdef RAW
  return sin(___a___);
#endif
#else
  return sin(___a___);
#endif
}

double CALL11(___Math______tan____D, double ___a___, double ___a___) {
#ifdef MULTICORE
  return 1;
#ifdef RAW
  return tan(___a___);
#endif
#else
  return tan(___a___);
#endif
}

double CALL11(___Math______acos____D, double ___a___, double ___a___) {
#ifdef MULTICORE
  return 1;
#ifdef RAW
  return acos(___a___);
#endif
#else
  return acos(___a___);
#endif
}

double CALL11(___Math______asin____D, double ___a___, double ___a___) {
#ifdef MULTICORE
  return 1;
#ifdef RAW
  return asin(___a___);
#endif
#else
  return asin(___a___);
#endif
}

double CALL11(___Math______atan____D, double ___a___, double ___a___) {
#ifdef MULTICORE
  return 1;
#ifdef RAW
  return atan(___a___);
#endif
#else
  return atan(___a___);
#endif
}

double CALL22(___Math______atan2____D_D, double ___a___, double ___b___, double ___a___, double ___b___) {
#ifdef MULTICORE
  return 1;
#ifdef RAW
  return atan2(___a___,___b___);
#endif
#else
  return atan2(___a___,___b___);
#endif
}

double CALL11(___Math______log____D, double ___a___, double ___a___) {
#ifdef MULTICORE
  return 1;
#ifdef RAW
  return log(___a___);
#endif
#else
  return log(___a___);
#endif
}

double CALL11(___Math______exp____D, double ___a___, double ___a___) {
#ifdef MULTICORE
  return 1;
#ifdef RAW
  return exp(___a___);
#endif
#else
  return exp(___a___);
#endif
}

double CALL11(___Math______sqrt____D, double ___a___, double ___a___) {
#ifdef MULTICORE
  return -1;
#ifdef RAW
  return sqrt(___a___);
#endif
#else
  return sqrt(___a___);
#endif
}

double CALL22(___Math______pow____D_D, double ___a___, double ___b___, double ___a___, double ___b___) {
#ifdef MULTICORE
  return 1;
#ifdef RAW
  return pow(___a___);
#endif
#else
  return pow(___a___,___b___);
#endif
}

double CALL11(___Math______ceil____D, double ___a___, double ___a___) {
#ifdef MULTICORE
  return 1;
#ifdef RAW
  return ceil(___a___);
#endif
#else
  return ceil(___a___);
#endif
}

double CALL11(___Math______floor____D, double ___a___, double ___a___) {
#ifdef MULTICORE
  return 1;
#ifdef RAW
  return floor(___a___);
#endif
#else
  return floor(___a___);
#endif
}

float CALL11(___Math______cosf____F, float ___a___, float ___a___) {
#ifdef MULTICORE
  return 1;
#ifdef RAW
  return cosf(___a___);
#endif
#else
  return cosf(___a___);
#endif
}

float CALL11(___Math______sinf____F, float ___a___, float ___a___) {
#ifdef MULTICORE
  return 1;
#ifdef RAW
  return sinf(___a___);
#endif
#else
  return sinf(___a___);
#endif
}

float CALL11(___Math______expf____F, float ___a___, float ___a___) {
#ifdef MULTICORE
  return 1;
#ifdef RAW
  return expf(___a___);
#endif
#else
  return expf(___a___);
#endif
}

float CALL11(___Math______sqrtf____F, float ___a___, float ___a___) {
#ifdef MULTICORE
  return 1;
#ifdef RAW
  return sqrtf(___a___);
#endif
#else
  return sqrtf(___a___);
#endif
}

float CALL11(___Math______logf____F, float ___a___, float ___a___) {
#ifdef MULTICORE
  return 1;
#ifdef RAW
  return logf(___a___);
#endif
#else
  return logf(___a___);
#endif
}

float CALL22(___Math______powf____F_F, float ___a___, float ___b___, float ___a___, float ___b___) {
#ifdef MULTICORE
  return 1;
#ifdef RAW
  return powf(___a___,___b___);
#endif
#else
  return powf(___a___,___b___);
#endif
}
#endif
